acceptance/tests/resource/package/common_package_name_in_different_providers.rb

```diff
@@ -41,6 +41,12 @@ def verify_absent(hosts, pkg)
   clean_rpm agent, rpm_options
   setup_rpm agent, rpm_options
   send_rpm agent, rpm_options
+
+  # DNF requires a cache reset to make local repositories accessible.
+  cmd = rpm_provider(agent)
+  if cmd == 'dnf'
+    on(agent, 'dnf clean metadata')
+  end
```

Comment:
Should this logic live in RpmUtils, or is the logic specific to this test?

Also meta question, does this affect puppet creating a repo and installing a package from the repo in the same run? Or is this only because we're creating a local repo for the test?